tests/ops/test_qubit_ops.py

```diff
@@ -0,0 +1,184 @@
+        observable_2_eigvecs = observable_pair[1][2]
+
+        qubit_unitary_2 = qml.Hermitian.diagonalizing_gates(observable_2, wires = [0])
+
+        key = tuple(observable_2.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], observable_2_eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], observable_2_eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary_2[0].params, observable_2_eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 2
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
+    @pytest.mark.usefixtures("tear_down_hermitian")
+    def test_hermitian_diagonalizing_gatesi_same_observable_twice(self, observable, eigvals, eigvecs, tol):
+        """Tests that the diagonalizing_gates method of the Hermitian class keeps the same dictionary entries upon multiple calls."""
+        qubit_unitary = qml.Hermitian.diagonalizing_gates(observable, wires = [0])
+
+        key = tuple(observable.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary[0].params, eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 1
+
+        qubit_unitary = qml.Hermitian.diagonalizing_gates(observable, wires = [0])
+
+        key = tuple(observable.flatten().tolist())
+        assert np.allclose(qml.Hermitian._eigs[key]["eigval"], eigvals, atol=tol, rtol=0)
+        assert np.allclose(qml.Hermitian._eigs[key]["eigvec"], eigvecs, atol=tol, rtol=0)
+
+        assert np.allclose(qubit_unitary[0].params, eigvecs.conj().T, atol=tol, rtol=0)
+        assert len(qml.Hermitian._eigs) == 1
+
+
+
+class TestQubitIntegration:
+    """Integration for the qubit based operations."""
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
+    @pytest.mark.usefixtures("tear_down_hermitian")
+    def test_hermitian_diagonalizing_gates_integration(self, observable, eigvals, eigvecs, tol):
+        """Tests that the diagonalizing_gates method of the Hermitian class contains contains a gate that diagonalizes the
+        given observable."""
+        num_wires = 2
+
+        obs = np.kron(observable, observable) 
+        eigvals = np.kron(eigvals, eigvals)
+
+        dev = qml.device('default.qubit', wires=num_wires)
+
+        diag_gates = qml.Hermitian.diagonalizing_gates(obs, wires=list(range(num_wires)))
+
+        assert len(diag_gates) == 1
+
+        U = diag_gates[0].parameters[0]
+        x = U @ obs @ U.conj().T
+        off_diagonal_indices = np.where(~np.eye(x.shape[0], dtype=bool))
+
+        assert np.all([np.isclose(elem, 0, atol=tol, rtol=0) for elem in x[off_diagonal_indices]])
+        assert np.allclose(np.sort(eigvals), np.sort(np.diag(x)), atol=tol, rtol=0)
```

Comment:
The above three lines can maybe be reduced to a single assert statement. This might be more succinct (and avoid needing to calculate the off-diagonal indices):
```python
assert np.allclose(np.diag(eigvals), x, atol=tol, rtol=0)
```
Essentially, just checking that `x` is the identical to the diagonal matrix of eigenvalues.

Note that no sorting should be required, as the order of the eigenvalues should already match the order of the eigenvectors making up `U`.

Reply:
That's a really nice one, thanks!

The need for sorting comes from the difference of how we store the eigenvalues and how numpy returns them. When calling ``qml.Hermitian.diagonalizing_gates(qml.PauliZ, wires=list(range(num_wires)))``, we get the numpy order e.g. for ``PauliZ`` we get ``[-1, 1]``, whereas we store it as ``[1, -1]``.

Reply:
Oh, good point!